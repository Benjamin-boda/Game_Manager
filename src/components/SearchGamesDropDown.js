import React, {Fragment} from "react";
import {Link} from "react-router-dom"

const SearchGamesDropDown = ({data, setData, setDataIsReady}) => {
    const onClick = () => {
        setData([])
        setDataIsReady(false)
    }
    return (
        <Fragment>
            <li key={data.title + data.platform}>
                <Link onClick={onClick} to={`/games/${data.title}/${data.platform}`}>
                    {data.title} - {data.platform}
                </Link>    
            </li>
        </Fragment>
    )
}

export const SearchGamesDropDownNoItem = () => {
    return (
        <Fragment>
            <li>
                <span>No result</span>
            </li>
        </Fragment> 
    )
}

export {SearchGamesDropDown as default}