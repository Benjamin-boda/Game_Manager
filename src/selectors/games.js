import moment from "moment";

//Get visible games

const getVisibleGames = (games, filters) => {
    return games.filter((game) => {
        const consoleMatch = game.platform.toLowerCase().includes(filters.console.toLowerCase());
        const textMatch = game.title.toLowerCase().includes(filters.text.toLowerCase());
        const statusMatch = game.played.toLowerCase().includes(filters.status.toLowerCase());

        return textMatch && statusMatch && consoleMatch;
    })
    .sort((a, b) => {
        if (filters.sortBy === "newDate") {
            return a.boughtAt < b.boughtAt ? 1 : -1;
        } else if (filters.sortBy === "oldDate") {
            return a.boughtAt < b.boughtAt ? -1 : 1;
        };
    })
};

export default getVisibleGames;