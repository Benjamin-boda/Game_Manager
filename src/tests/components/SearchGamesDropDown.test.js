import { mount, shallow } from "enzyme";
import SearchGamesDropDown, {SearchGamesDropDownNoItem} from "../../components/SearchGamesDropDown";
import games from "../fixtures/games";
import { Provider, useSelector, useDispatch } from "react-redux";
import React from "react";
import thunk from 'redux-thunk'
import configureMockStore from "redux-mock-store";
import { shallowToJson } from "enzyme-to-json";
import { MemoryRouter, Switch } from 'react-router-dom'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

test("should render SearchGamesDropDown page correctly", () => {
    const data = {title: "sonic", platform: "ps4"}
    const wrapper = shallowToJson(mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/']} keyLength={0}>
                <SearchGamesDropDown data={data}
                    /> 
            </MemoryRouter>
                
            
        </Provider>
        ))
    expect(wrapper).toMatchSnapshot();
});

test("should render SearchGamesDropDownNoItem page correctly", () => {
    const wrapper = shallowToJson(mount(
        <Provider store={store}>
            
                <SearchGamesDropDownNoItem 
                /> 
            
        </Provider>
        ))
    expect(wrapper).toMatchSnapshot();
});