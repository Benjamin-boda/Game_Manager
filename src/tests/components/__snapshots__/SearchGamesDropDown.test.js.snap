// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchGamesDropDown page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SearchGamesDropDown
        data={
          Object {
            "platform": "ps4",
            "title": "sonic",
          }
        }
      >
        <li
          key="sonicps4"
        >
          <Link
            to="/games/sonic/ps4"
          >
            <LinkAnchor
              href="/games/sonic/ps4"
              navigate={[Function]}
            >
              <a
                href="/games/sonic/ps4"
                onClick={[Function]}
              >
                sonic
                 - 
                ps4
              </a>
            </LinkAnchor>
          </Link>
        </li>
      </SearchGamesDropDown>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`should render SearchGamesDropDownNoItem page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchGamesDropDownNoItem>
    <li>
      <span>
        No result
      </span>
    </li>
  </SearchGamesDropDownNoItem>
</Provider>
`;
